### PR TITLE
refactor(crash reporting): add crash reporting using sentry

### PR DESCRIPTION
**Pull request by subotic**

### CODE REVIEW

File: ext/sentry/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+endif()
+
+
+ExternalProject_Add(project_sentry
+        DEPENDS curl
+        INSTALL_DIR ${COMMON_LOCAL}
+        URL https://github.com/getsentry/sentry-native/releases/download/0.6.6/sentry-native.zip
+        DOWNLOAD_DIR ${COMMON_SRCS}/downloads
+        ${timestamp_policy}
+        #DOWNLOAD_EXTRACT_TIMESTAMP NEW
```
Comment (BalduinLandolt) [inline ext/sentry/CMakeLists.txt:21]: needed?

File: src/sipi.cpp [hunk 2]
```diff
@@ -672,6 +673,19 @@ int main(int argc, char *argv[]) {
                        "Logging level Value can be: 'DEBUG', 'INFO', 'WARNING', 'ERR', 'CRIT', 'ALERT', 'EMERG'.")
             ->transform(CLI::CheckedTransformer(logLevelMap, CLI::ignore_case))->envname("SIPI_LOGLEVEL");
 
+    // sentry configuration
+    // sentry_dsn
+    // sentry_release
+    // sentry_environment
```
Comment (BalduinLandolt) [inline src/sipi.cpp:679]: needed?

**Review by BalduinLandolt — APPROVED** [review]
I have to approve this pretty blindly, but it looks reasonable. There are some comments you could probably remove.

File: ext/sentry/CMakeLists.txt (hunk 1 above)
Comment (subotic) [reply ext/sentry/CMakeLists.txt]: removed

File: src/sipi.cpp (hunk 2 above)
Comment (subotic) [reply src/sipi.cpp]: thanks. removed.

**Pull request merged**